### PR TITLE
Show correct decimals on product price and show the sale price of products

### PR DESCRIPTION
**Pull request by raja651**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes 18825-gh-Automattic/woocommerce.com and 21785-gh-Automattic/woocommerce.com.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure your WCCOM environment is running on this [PR](21705-gh-Automattic/woocommerce.com) branch 
2. Change product prices to have some decimal values for couple of products  on your WCCOM local and make couple of products on sale
3. clear cache of your WCCOM local `npm run wp:cli cache flush`
4. Next you'll need to point your WC Core environment to your local WCCOM by following instruction on this PR https://github.com/woocommerce/woocommerce/pull/51170
5. Go to  [in-app marketplace](http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions).
6. view extensions and Discover tab
7. confirm the products price with decimal values are displayed correctly
8. Also confirm the sale price is shown similar to what we have on WCCOM
Screenshot 2024-09-18 at 12 58 08 PM
Screenshot 2024-09-18 at 12 58 02 PM

9. Switch VO on mac and confirm the product prices on card are read out correctly
10. Bonus: Test this change doesn't break anything on IAM for older versions

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
Show the sale price and display the correct number of decimals for the product card price.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Review by KokkieH — COMMENTED** [review]
I'm having trouble connecting to the featured endpoint locally atm. However, for Extensions/Search I'm not seeing the sale price:

Extensions_‹_WooCommerce_‹_WooCommerce_Core_E2E_Test_Suite
Extensions_‹_WooCommerce_‹_WooCommerce_Core_E2E_Test_Suite

Prices with decimals seem to be fine, though:

Extensions_‹_WooCommerce_‹_WooCommerce_Core_E2E_Test_Suite

**Comment by raja651** [comment]
> I'm having trouble connecting to the featured endpoint locally atm. However, for Extensions/Search I'm not seeing the sale price:

Fixed a typo [a6bb6f0](https://github.com/woocommerce/woocommerce/pull/51499/commits/a6bb6f0947346d214fabd67a28bd35a92072250f) 🤦

**Review by KokkieH — APPROVED** [review]
I'm still unable to connect to my local WCCOM, so could only check against data being returned by the production API for the Discover tab, and there aren't currently any products that are on sale or that have decimals in the price.

However, on search I've confirmed decimals and sale prices are working as expected, and sale prices are being properly read out by VO.

Unrelated, I noticed the announcements for the review data aren't great, but I'll create a separate task to address that.

File: plugins/woocommerce-admin/client/marketplace/components/product-card/product-card-footer.tsx [hunk 1]
```diff
@@ -144,6 +144,32 @@ function ProductCardFooter( props: { product: Product } ) {
 		return '';
 	}
 
+	const getReaderPriceLabel = () => {
+		if ( product.isOnSale ) {
+			return sprintf(
+				//translators: %1$s is the sale price of the product, %2$s is the regular price of the product, %3$s is the billing period
+				__( 'Sale Price %1$s, %2$s %3$s', 'woocommerce' ),
```
Comment (andfinally) [inline plugins/woocommerce-admin/client/marketplace/components/product-card/product-card-footer.tsx]: I found the announcement of the regular price confusing, might help to spell it out.

```suggestion
				__(
					'Sale Price %1$s  %3$s, regular price %2$s %3$s',
					'woocommerce'
				),
```

**Review by andfinally — APPROVED** [review]
Thanks, this will be super handy for our upcoming sale! I'm also getting `cURL error 7 ... Connection refused` from the featured tab API request. I tested by copying the JSON from my local endpoint and just echoing it from `WC_Helper_Admin::get_featured`, and the price rendered as expected, same in search results.

I just left a small suggestion for the screen reader output.

File: plugins/woocommerce-admin/client/marketplace/components/product-card/product-card-footer.tsx (hunk 1 above)
Comment (raja651) [reply plugins/woocommerce-admin/client/marketplace/components/product-card/product-card-footer.tsx]: done in [f121b87](https://github.com/woocommerce/woocommerce/pull/51499/commits/f121b8711ec22b1e7e680700186fdd6e0ab12576)

**Pull request merged**